server/game/drawcard.js

```diff
@@ -354,22 +353,38 @@ class DrawCard extends BaseCard {
         this.fate = Math.max(0, this.fate + amount);
     }
 
+    setPersonalHonor(token) {
+        this.personalHonor = token || null;
+    }
+
+    get isHonored() {
+        return !!this.personalHonor && !!this.personalHonor.honored;
+    }
+
     honor() {
         if(this.isDishonored) {
-            this.isDishonored = false;
+            this.makeOrdinary();
         } else {
-            this.isHonored = true;
+            this.setPersonalHonor(new StatusToken(this.game, this, true));
         }
     }
 
+    get isDishonored() {
+        return !!this.personalHonor && !!this.personalHonor.dishonored;
+    }
+
     dishonor() {
         if(this.isHonored) {
-            this.isHonored = false;
+            this.makeOrdinary();
         } else {
-            this.isDishonored = true;
+            this.setPersonalHonor(new StatusToken(this.game, this, false));
         }
     }
 
+    makeOrdinary() {
+        this.setPersonalHonor();
+    }
```

Comment:
I think it may be worth setting `card` to `null` on any personalHonor that may already be present on the card so that if the token changes before resolution then the `.token.card` would be `null`.

Reply:
I don't really want to set `card` to `null` as it loses information.  I've added tests which check that the token in the event is still the token on the card, which I think achieves the same aim.